README.md

```diff
@@ -17,26 +18,30 @@ The Si702x should be connected as follows:
 To instantiate a new Si702x object you need to pass in a preconfigured I&sup2;C object and an optional I&sup2;C base address. If no base address is supplied, the default address of `0x80` will be used.
 
 ```squirrel
-hardware.i2c12.configure(CLOCK_SPEED_100_KHZ)
-tempHumid <- Si702x(hardware.i2c12)
+hardware.i2c12.configure(CLOCK_SPEED_400_KHZ)
+tempHumid <- Si702x(hardware.i2c89)
 ```
 
 ### Class Methods
 
-### readTemp()
+#### read([callback])
 
-The **readTemp()** method returns the temperature in degrees Celsius:
+The **read()** method takes an optional callback for asynchronous operation. The callback should take one parameter: a results table that contains the following:
 
-```squirrel
-server.log(tempHumid.readTemp() + "C")
-```
-
-### readHumidity()
+| Key         | Type  | Description           |
+| ----------- | ----- | --------------------- |
+| temperature | float | Temperature (°C)      |
+| humidity    | float | Relative humidity (%) |
 
-The **readHumidity()** function returns the relative humidity (0-100 per cent):
+If the callback is null or omitted, the method will return the above table to the caller instead.
 
+### Example Usage
 ```squirrel
-server.log(tempHumid.readHumidity() + "%")
+temphumid.read(function(result) {
+    if (result) {
+        server.log(format("Temperature: %.01f°C, Relative Humidity: %.01f%%", result.temperature, result.humidity));
```

Comment:
Yeah, that's totally fine. You've programmed ```read``` to invoke the callback, so no problems there.